File: front/ui/src/app/data/CompletedGame.ts
import { Difficulty } from './Difficulty';
import { GameStep } from './GameStep';
import { Maze, MazeResponse } from './FieldContent';

export interface CompletedGame {
  difficulty: Difficulty,
  steps: Array<GameStep>,
  userId: string,
  totalTimeInMilliseconds: number,
  startTime: number,
  maze: MazeResponse
}

File: front/service/src/main/java/com/coinhunt/front/data/CompletedGame.java
package com.coinhunt.front.data;

import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.ToString;

@AllArgsConstructor
@Getter
@ToString
public class CompletedGame {

	private final Difficulty difficulty;

	private final List<GameStep> steps;

	private final String userId;

	private final long totalTimeInMilliseconds;

	private final long startTime;

	private final MazeWrapper maze;
}
